src/compute-client/src/controller/instance.rs

```diff
@@ -897,13 +908,18 @@ where
         // Install a compaction hold on `id` at `timestamp`.
         let mut updates = BTreeMap::new();
         updates.insert(id, ChangeBatch::new_from(timestamp.clone(), 1));
-        self.update_read_capabilities(&mut updates);
+        match &peek_target {
```

Comment:
Yeah... this may be a wise thing to revisit as the dust settles. (`ActiveInstance::update_read_capabilities` makes some effort to pass things through to the storage controller, though IIUC this is ambiguous in the case where a collection both has some compute and storage representation... it's possible that we can make that more robust and avoid the branch here, but this felt like the conservative / local change to make for now.)